provider/fs/src/fs_data_provider.rs

```diff
@@ -72,8 +72,24 @@ impl FsDataProvider {
         if !path.exists() {
             return Err(DataErrorKind::MarkerNotFound.with_req(marker, req));
         }
-        if !req.id.marker_attributes.is_empty() {
-            path.push(req.id.marker_attributes.as_str());
+        'attr: {
+            if !req.id.marker_attributes.is_empty() {
+                if req.metadata.attributes_prefix_match {
+                    for candidate in std::fs::read_dir(&path)? {
```

Comment:
Question: does `std::fs::read_dir` read in alphabetical order? If so, please add a comment about that. If not, we need to walk the whole dir and select the first alphabetically.